### DISCUSSION TITLE
File selection/testing with Playwright

### DISCUSSION THREAD

**Discussion by absreim** [react-dropzone/react-dropzone#1339] [General]
I recently set out to write an end-to-end test using Playwright for a side project that I am working on. Part of the test involves selecting a file via the browser's native file dialog. The dialog can be opened by clicking on the React Dropzone component.

I tried to listen for the "filechooser" event via the instructions at https://playwright.dev/docs/input#upload-files, but Playwright was not detecting the "filechooser" event for some reason.

After doing some research, it appears that the event does not work with the File System Access API. I was able to get the "filechooser" event to fire by setting React Dropzone to disable the File System Access API:

`useDropzone({useFsAccessApi: false})`

I figured I would post this finding here in case someone else runs into the same problem in the future. 😀

**Comment by AlonLavy** [comment]
I love you

**Comment by neverased** [comment]
I've spent the last two days looking for a reliable fix for Chrome's failure on this, and your solution worked like a charm. Big thanks!

**Comment by rolandjitsu** [comment]
@absreim thanks for posting this useful information.

**Comment by luna215** [comment]
wow...thank you so much!

**Comment by st-manhle** [comment]
You save my day 🙌🏻🙌🏻🙌🏻

**Comment by ewarrenG** [comment]
This isn't working for me... :(

In the `useDropzone` hook, I also have 
```
    onDrop,
    disabled: disableInput,
```

might these conflict? Is there something else I might be missing? Thanks in advance

**Comment by stefan-girlich** [comment]
Thanks for documenting this; I couldn't have figured it out by myself.

Root issue:
https://github.com/microsoft/playwright/issues/18267
